src/cc/bcc_syms.cc

```diff
@@ -100,12 +100,38 @@ void ProcSyms::refresh() {
 
 int ProcSyms::_add_module(const char *modname, uint64_t start, uint64_t end,
                           void *payload) {
+  struct ns_cookie nsc = {-1, -1};
+  bool ns_switch = false;
+  int arc;
   ProcSyms *ps = static_cast<ProcSyms *>(payload);
   auto it = std::find_if(ps->modules_.begin(), ps->modules_.end(),
                [=](const ProcSyms::Module &m) { return m.name_ == modname; });
-  if (it == ps->modules_.end())
-    it = ps->modules_.insert(ps->modules_.end(), modname);
+  if (it == ps->modules_.end()) {
+    // If modname references a perf-map, determine if we need to enter a mount
+    // namespace in order to read symbols from it later.
+    if (strstr(modname, ".map") != nullptr) {
+      ns_switch = bcc_procutils_enter_mountns(ps->pid_, &nsc);
+      if (ns_switch) {
+        char new_modname[4096];
+        arc = access(modname, R_OK);
+        bcc_procutils_exit_mountns(&nsc);
+
+        if (arc != 0) {
+          snprintf(new_modname, sizeof (new_modname), "/tmp/perf-%d.map",
```

Comment:
@kmjohansen: Quick question here. Is it possible that even in the new mount namespace, we should also use [NS-PID](https://github.com/iovisor/bcc/blob/master/src/cc/bcc_perf_map.c#L23) instead of just `ps->pid_`?

Also, I don't get when would this part of logic actually be hit. i.e., access to `modname` fails while access `/tmp/perf-PID.map` works. In fact, to my understanding `chroot` can also happen in combine with mount namespace, so `modname` should actually be better since it is the absolute path taking the Process's root into consideration.